aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/AwsMetricAttributeGenerator.java

```diff
@@ -152,35 +212,132 @@ private static void setEgressOperation(SpanData span, AttributesBuilder builder)
    * and RPC attributes to use here, but this is a sufficient starting point.
    */
   private static void setRemoteServiceAndOperation(SpanData span, AttributesBuilder builder) {
+    String remoteService = UNKNOWN_REMOTE_SERVICE;
+    String remoteOperation = UNKNOWN_REMOTE_OPERATION;
     if (isKeyPresent(span, AWS_REMOTE_SERVICE) || isKeyPresent(span, AWS_REMOTE_OPERATION)) {
-      setRemoteService(span, builder, AWS_REMOTE_SERVICE);
-      setRemoteOperation(span, builder, AWS_REMOTE_OPERATION);
+      remoteService = getRemoteService(span, AWS_REMOTE_SERVICE);
+      remoteOperation = getRemoteOperation(span, AWS_REMOTE_OPERATION);
     } else if (isKeyPresent(span, RPC_SERVICE) || isKeyPresent(span, RPC_METHOD)) {
-      setRemoteService(span, builder, RPC_SERVICE);
-      setRemoteOperation(span, builder, RPC_METHOD);
+      remoteService = getRemoteService(span, RPC_SERVICE);
+      remoteOperation = getRemoteOperation(span, RPC_METHOD);
     } else if (isKeyPresent(span, DB_SYSTEM) || isKeyPresent(span, DB_OPERATION)) {
-      setRemoteService(span, builder, DB_SYSTEM);
-      setRemoteOperation(span, builder, DB_OPERATION);
-    } else if (isKeyPresent(span, FAAS_INVOKED_PROVIDER) || isKeyPresent(span, FAAS_INVOKED_NAME)) {
-      setRemoteService(span, builder, FAAS_INVOKED_PROVIDER);
-      setRemoteOperation(span, builder, FAAS_INVOKED_NAME);
+      remoteService = getRemoteService(span, DB_SYSTEM);
+      remoteOperation = getRemoteOperation(span, DB_OPERATION);
+    } else if (isKeyPresent(span, FAAS_INVOKED_NAME) || isKeyPresent(span, FAAS_TRIGGER)) {
+      remoteService = getRemoteService(span, FAAS_INVOKED_NAME);
+      remoteOperation = getRemoteOperation(span, FAAS_TRIGGER);
     } else if (isKeyPresent(span, MESSAGING_SYSTEM) || isKeyPresent(span, MESSAGING_OPERATION)) {
-      setRemoteService(span, builder, MESSAGING_SYSTEM);
-      setRemoteOperation(span, builder, MESSAGING_OPERATION);
+      remoteService = getRemoteService(span, MESSAGING_SYSTEM);
+      remoteOperation = getRemoteOperation(span, MESSAGING_OPERATION);
```

Comment:
I thought this would be: `{messaging.destination.name} {messaging.operation}`

Reply:
thanks for calling it out. I am thinking moving `{messaging.destination.name}` to `aws.remote.target` attribute later and keep `remote.operation` with `messaging.operation`, what do you think?